packages/babel-core/src/transformation/normalize-file.js

```diff
@@ -35,9 +34,8 @@ export default function* normalizeFile(
       throw new Error("AST root must be a Program or File node");
     }
 
-    const { cloneInputAst } = options;
-    if (cloneInputAst) {
-      ast = cloneDeep(ast);
+    if (options.cloneInputAst) {
+      ast = JSON.parse(JSON.stringify(ast));
```

Comment:
JSON stringify behaves in different way, it does not respect / clone correctly regexp values, and other complex types

Reply:
I don't believe we produce any values besides primitives, simple objects, and arrays.

Reply:
`Literal` node can have value of bigint and regexp

Reply:
The value of BigIntLiteral is a bigint.

Reply:
`@babel/core` is not compatible with ESTree anyway so it's not a big problem, but when we'll drop support for Babel 8 versions we might decide to use antive bigint literal for values.

Does Node.js expose the structured clone algorithm used by `postMessage`?

Reply:
Ok it looks like there is `v8.deserialize(v8.serialize(original));`, but I have to check if it's correctly polyfilles when bundling for the browser.

Reply:
I feel weird importing a node-specific module. Do we have a similar node/browser style conditional import?

We could implement t this as `JSON` with a replacer and reviver. Or, we could add a new `t.cloneDeepAndCloneLoc`?

Reply:
We have [a few files](https://github.com/babel/babel/blob/1ef78ca55b39773cbb5974960b7b34e2fca6e043/packages/babel-core/package.json#L44-L46) with Node-specific implementations.

I think here a replacer/reviver which adds support for bigints would be enough if you prefer to keep a single implementation.

Reply:
Thought cloneNode supports both deep/loc options (just since I happened to be looking at it recently) https://github.com/babel/babel/blob/2ae19d01b132f5222e1d5bee2c83921e2f107d70/packages/babel-types/src/clone/cloneNode.ts#L28-L31?

Reply:
It copies the same `loc` object, we need to clone it.

Reply:
Ah right

Would this be like what we did in https://github.com/babel/babel/pull/10858 then?

Reply:
Done, thanks for the pointers everyone!